src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java

```diff
@@ -0,0 +1,98 @@
+package com.thealgorithms.maths;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import org.junit.jupiter.api.Test;
+
+public class ConvolutionFFTTest {
+
+    /**
+     * Helper method to create a complex signal from an array of doubles.
+     */
+    private ArrayList<FFT.Complex> createComplexSignal(double[] values) {
+        ArrayList<FFT.Complex> signal = new ArrayList<>();
+        for (double value : values) {
+            signal.add(new FFT.Complex(value, 0));
+        }
+        return signal;
+    }
+
+    /**
+     * Helper method to compare two complex signals for equality within a small margin of error.
+     */
+    private void assertComplexArrayEquals(ArrayList<FFT.Complex> expected, ArrayList<FFT.Complex> result, double delta) {
+        assertEquals(expected.size(), result.size(), "Signal lengths are not equal.");
+        for (int i = 0; i < expected.size(); i++) {
+            FFT.Complex expectedValue = expected.get(i);
+            FFT.Complex resultValue = result.get(i);
+            assertEquals(expectedValue.real(), resultValue.real(), delta, "Real part mismatch at index " + i);
+            assertEquals(expectedValue.imaginary(), resultValue.imaginary(), delta, "Imaginary part mismatch at index " + i);
+        }
+    }
+
+    @Test
```

Comment:
Will be better to use `@ParameterizedTest` for this and additional class to store data.

Reply:
Done!